kratos/python_scripts/orchestrators/orchestrator.py

```diff
@@ -0,0 +1,186 @@
+                analysis_stage_class_name = analysis_stage_reg_entry["ClassName"]
+                if "ModuleName" in analysis_stage_reg_entry:
+                    # Standard case in which the analysis stage is registered in a module
+                    analysis_stage_module_name = analysis_stage_reg_entry["ModuleName"]
+                else:
+                    # Alternative in which only the "ClassName" is provided with no module (e.g. custom user-defined stages in MainKratos.py)
+                    analysis_stage_module_name = None
+            else:
+                err_msg = f"Trying to retrieve a non-registered analysis stage '{input_analysis_stage}'"
+                raise ValueError(err_msg)
+        else:
+            # Throw a warning to let the user know that the current analysis stage is not registered
+            warn_msg = f"Current analysis stage '{input_analysis_stage}' is not registered. Assuming that provided string contains module and class names."
+            KratosMultiphysics.Logger.PrintWarning(warn_msg)
+
+            # Get the Python module name and class name implementing each analysis stage
+            # Note that we assume that the class name is the provided module (file) name in CamelCase
+            analysis_stage_module_name = input_analysis_stage
+            analysis_stage_class_name = analysis_stage_module_name.split('.')[-1]
+            analysis_stage_class_name = ''.join(x.title() for x in analysis_stage_class_name.split('_'))
+
+        # Import the stage module
+        # Note that for the case of custom analysis stage (i.e. no Kratos module) we check if the class exists in the __main__ module
+        if analysis_stage_module_name is not None:
+            analysis_stage_module = importlib.import_module(analysis_stage_module_name)
+        else:
+            analysis_stage_module = sys.modules["__main__"]
+
+        # Create the analysis stage instance
+        input_stage_settings = KratosMultiphysics.Parameters(current_stage_settings)
+        if hasattr(analysis_stage_module, analysis_stage_class_name):
+            # First we check for the expected class name
+            analysis_stage_class = getattr(analysis_stage_module, analysis_stage_class_name)
+            stage_instance = analysis_stage_class(self.__project.GetModel(),  input_stage_settings)
+        elif hasattr(analysis_stage_module, "Create"):
+            # If Kratos convention is not fulfilled we search for a Create method
+            stage_instance = analysis_stage_module.Create(self.__project.GetModel(), input_stage_settings)
+        else:
+            err_msg = f"Analysis stage in '{analysis_stage_module_name}' Python module cannot be created. Please check class name or provide a 'Create' method."
+            raise Exception(err_msg)
+
+        return stage_instance
+
+    def RunCurrentStagePreprocess(self, stage_name: str, data: typing.Optional[dict] = None):
```

Comment:
```suggestion
    def RunCurrentStagePreprocess(self, stage_name: str, data: typing.Optional[typing.Dict] = None):
```

pretty internal types can only be used from 3.9
then you could use it even better: `data: dict | None = None`